pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/StarTreeIndexReader.java

```diff
@@ -121,7 +124,9 @@ private void mapBufferEntries(int starTreeId,
           _dataBuffer, ByteOrder.BIG_ENDIAN));
     }
     // Load metric (function-column pair) forward indexes
-    for (AggregationFunctionColumnPair functionColumnPair : starTreeMetadata.getFunctionColumnPairs()) {
+    TreeMap<AggregationFunctionColumnPair, AggregationSpec> deduplicatedAggregationSpecs =
```

Comment:
Can we do this at `StarTreeV2Metadata` loading time (in the constructor), essentially keeping the stored `AggregationFunctionType` within `AggregationFunctionColumnPair`

Reply:
I went back and forth about how best to do this as detailed in the issue and PR description. A consequence of de-duplicating in the `StarTreeV2Metadata` and the `StarTreeV2BuilderConfig` is that the metadata for a segment will only contain the stored aggregation function types, and not any superfluous aggregation function types.  This has implications for whether a segment should be rebuilt when compared to the existing config (see `shouldModifyExistingStarTrees` in `StarTreeBuilderUtils`).  If both the metadata and builder config reflect the stored aggregated types then this should behave correctly.

Finally, it might be surprising to the end user if some of their StarTree aggregation configuration does not actually land up in metadata - I was considering adding a warning or validation to the TableConfig.   What do you think?

Reply:
Within the in-memory loaded metadata (not the on disk copy which might already be generated) and the builder config, we can always keep the stored aggregates, and the query time lookup should also be converted to stored type first. This way it should work as expected.
I wouldn't expect user trying to match the segment metadata with table config, but +1 on logging warnings when the non-stored aggregates are provided within the table config

Reply:
I've not found a way to log warnings, but have validated that the configuration does not contain duplicates in this commit: https://github.com/apache/pinot/pull/12164/commits/ab6b03d58955b2b748ac801641b4f629d9b58152